### PR TITLE
chore(deps): update to webpack 2.1.0-beta.20 and typescript 2.0.0

### LINKED ISSUE #130: update to webpack 2 and typescript 2

### PR DESCRIPTION
**Pull request by manekinekko**
update to webpack 2.1.0-beta.20 and typescript 2.0.0
fix #130

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,9 +50,9 @@
     "rimraf": "^2.5.2",
     "source-map-loader": "^0.1.5",
     "ts-loader": "^0.8.2",
-    "typescript": "^1.8.9",
+    "typescript": "^2.0.0",
     "typings": "~1.0.5",
-    "webpack": "^1.13.1",
+    "webpack": "^2.1.0-beta.20",
```
Comment (gdi2290) [inline package.json]: does `beta-20` work?

Comment (gdi2290) [inline package.json]: nvm I was thinking of typescript

File: package.json [hunk 2]
```diff
@@ -50,9 +50,9 @@
     "rimraf": "^2.5.2",
     "source-map-loader": "^0.1.5",
     "ts-loader": "^0.8.2",
-    "typescript": "^1.8.9",
+    "typescript": "^2.0.0",
```
Comment (gdi2290) [inline package.json]: add `-beta`

File: package.json [hunk 3]
```diff
@@ -50,9 +50,9 @@
...
     "typings": "~1.0.5",
```
Comment (gdi2290) [inline package.json]: remove typings for @types

File: package.json (hunk 1 above)
Comment (manekinekko) [inline package.json]: yes it does. Otherwise, I wouldn't send this PR :)

File: package.json (hunk 2 above)
Comment (manekinekko) [inline package.json]: `npm i typescript@beta -D` is creating ` "typescript": "^2.0.0", `  in package.json. Is that what you want?

File: package.json (hunk 3 above)
Comment (manekinekko) [inline package.json]: I switched to @types:
```
"@types/body-parser": "0.0.29",
    "@types/compression": "0.0.29",
    "@types/cookie-parser": "^1.3.29",
    "@types/es6-shim": "0.0.28",
    "@types/express": "^4.0.29",
    "@types/express-serve-static-core": "^4.0.29",
    "@types/mime": "0.0.28",
    "@types/node": "^4.0.30",
    "@types/serve-static": "^1.7.27",
````
I updated the `tsconfig` accordingly:

```
"include": [
    "node_modules/@types/**/*.d.ts",
    "src/**/*.ts"
  ],
  "exclude": [
    "node_modules"
  ],
```
But I'm getting so many errors related to types: `error TS2304: Cannot find name 'Map'.`

I'll look at this one later.

**Comment by Igonato** [comment]
Any plans on getting tree-shaking to work?

**Comment by MarkPieszak** [comment]
I haven't tried to get it with webpack2 just yet but looks like it *should* be available in TS 2beta? Not positive. Something to do with target es5 module es6. Here's the PR (https://github.com/Microsoft/TypeScript/pull/9042)

**Pull request merged**